srv/salt/_modules/osd.py

```diff
@@ -1455,22 +1472,44 @@ def remove(self):
 
         if self.force:
             log.warning("Forcing OSD removal")
-        else:
-            self.empty()
 
-        # Terminate
-        self.terminate()
+            # Terminate
+            self.terminate()
 
-        # Unmount filesystems
-        result = self.unmount()
-        if result:
-            return result
+            # Best effort depending on the reason for the forced removal
+            self.mark_destroyed()
+            update_destroyed(self._osd_disk(), self.osd_id)
+        else:
+            for func in [self.empty, self.terminate]:
+                msg = func()
+                if msg:
+                    log.error(msg)
+                    return msg
+
+            # Inform Ceph
+            #
+            # Consider this a hard requirement for graceful removal. If the
+            # OSD cannot be marked and recorded, stop the process.
+            if self.mark_destroyed():
+                msg = update_destroyed(self._osd_disk(), self.osd_id)
+                if msg:
+                    log.error(msg)
+                    return msg
+                log.debug("OSD {} marked and recorded".format(self.osd_id))
```

Comment:
That could be a log.info/warning ( something that a user sees on his screen )

Reply:
We are in a module, so the message will not come back to the admin on the Salt master.  I'll set it to info on the off chance that somebody runs this as a salt-call.